README.md

```diff
@@ -240,6 +282,52 @@ gpio partition string input.parquet output/ --column mgrs_code --chars 2
 gpio partition string input.parquet output/ --column region --chars 1 --hive
 ```
 
+#### partition h3
+
+Partition a GeoParquet file by H3 hexagonal cells at a specified resolution. Automatically adds H3 column if it doesn't exist.
+
+```
+$ gpio partition h3 --help
+Usage: gpio partition h3 [OPTIONS] INPUT_PARQUET [OUTPUT_FOLDER]
+
+  Partition a GeoParquet file by H3 cells at specified resolution.
+
+  Creates separate GeoParquet files based on H3 cell prefixes at the
+  specified resolution. If the H3 column doesn't exist, it will be
+  automatically added before partitioning.
+
+  Use --preview to see what partitions would be created without actually
+  creating files.
+
+Options:
+  --h3-name TEXT           Name of H3 column to partition by (default:
+                           h3_cell)
+  --resolution INTEGER     H3 resolution for partitioning (0-15, default: 9)
```

Comment:
So I just tested this with a 66mb file and it looks like I got 6000+ parquet files and it took 20 minutes. We could consider lowering the default. But I think the main thing to do is to do more automatic / dynamic things here. #10 likely covers most of this, but once we have that mechanism in there then I think we probably calculate the default automatically for the data input, based on [best practices](https://github.com/opengeospatial/geoparquet/blob/main/format-specs/distributing-geoparquet.md)

Reply:
We also might proactively put in a warning if it's going to generate a ton of small files. Like right now we say:

```
% gpio partition h3 c-america.parquet c-h3        
Processing 722,109 features...
Successfully added H3 column 'h3_cell' (resolution 9) to: /var/folders/s6/ctf1vwb95cnb44vtlbg4r6c40000gp/T/h3_enriched_c-america.parquet
Partitioning by H3 cells at resolution 9 (column: 'h3_cell')
```

Could just report out how many unique partitions it'll be partitioned to.

Also we might have the default 'partition' be lower than the default 'add' - then if people want to partition further or search by it it's already there.